File: bcube-services/payment-service/src/main/java/com/bcube/paymentservice/service/dto/response/PaymentResponse.java
package com.bcube.paymentservice.service.dto.response;

import com.bcube.paymentservice.persistance.entity.PaymentStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PaymentResponse {
    private Long id;
    private Long bookingId;
    private Long userId;
    private Long studioId;
    private Integer hourlyRateCentsSnapshot;
    private BigDecimal durationHours;
    private Integer baseAmountCents;
    private Integer discountAmountCents;
    private Integer finalAmountCents;
    private String currency;
    private PaymentStatus status;
    private String clientSecret;
    private Integer refundedAmountCents;
    private String studioName;
    private LocalDate bookingDate;
    private Instant createdAt;
    private String invoiceNumber;
}

File: bcube-web/src/app/models/responses/payment/payment-response.ts
export type PaymentStatus =
    | 'PENDING'
    | 'REQUIRES_PAYMENT'
    | 'REQUIRES_CARD_VERIFICATION'
    | 'SUCCEEDED'
    | 'FAILED'
    | 'FREE'
    | 'REFUNDED'
    | 'PARTIALLY_REFUNDED';

export interface PaymentResponse {
    id: number;
    bookingId: number;
    userId: number;
    studioId: number;
    hourlyRateCentsSnapshot: number;
    durationHours: number;
    baseAmountCents: number;
    discountAmountCents: number;
    finalAmountCents: number;
    currency: string;
    status: PaymentStatus;
    clientSecret: string | null;
    refundedAmountCents: number;
    studioName: string | null;
    bookingDate: string | null;
    createdAt: string;
    invoiceNumber: string | null;
}
